cluster/src/test/java/com/hartwig/pipeline/tertiary/orange/OrangeTest.java

```diff
@@ -66,38 +73,63 @@ protected List<String> expectedInputs() {
+                        + "-cuppa_chart_plot /data/input/tumor.cuppa.chart.png "
+                        + "-chord_prediction_txt /data/input/tumor_chord_prediction.txt "
+                        + "-peach_genotype_tsv /data/input/tumor.peach.genotype.tsv "
+                        + "-sigs_allocation_tsv /data/input/tumor.sig.allocation.tsv "
+                        + "-annotated_virus_tsv /data/input/tumor.virus.annotated.tsv "
+                        + "-pipeline_version_file /data/input/orange_pipeline.version.txt "
+                        + "-cohort_mapping_tsv /opt/resources/orange/cohort_mapping.tsv "
+                        + "-cohort_percentiles_tsv /opt/resources/orange/cohort_percentiles.tsv "
+                        + "-driver_gene_panel_tsv /opt/resources/gene_panel/37/DriverGenePanel.37.tsv "
+                        + "-known_fusion_file /opt/resources/fusions/37/known_fusion_data.37.csv "
+                        + "-ensembl_data_directory /opt/resources/ensembl_data_cache/37/ " + "-convert_germline_to_somatic";
         String cuppaFile = " -cuppa_feature_plot /data/input/tumor.cup.report.features.png";
-        String fileExistsCommand = "if [ -e /data/input/tumor.cup.report.features.png ]; then " + jarRunCommand + cuppaFile + " ; else " + jarRunCommand + " ; fi";
-        return Arrays.asList("mkdir -p /data/input/linx/plot", "echo '5.33' | tee /data/input/orange_pipeline.version.txt", fileExistsCommand);
+        String fileExistsCommand =
+                "if [ -e /data/input/tumor.cup.report.features.png ]; then " + jarRunCommand + cuppaFile + " ; else " + jarRunCommand
+                        + " ; fi";
+        return Arrays.asList("mkdir -p /data/input/linx/plot",
+                "echo '5.33' | tee /data/input/orange_pipeline.version.txt",
+                fileExistsCommand);
+    }
+
+    @Test
+    public void shouldAddResearchDisclaimerWhenResearchContext() {
+        Orange victim = new Orange(TestInputs.tumorMetricsOutput(),
+                TestInputs.referenceMetricsOutput(),
+                TestInputs.tumorFlagstatOutput(),
+                TestInputs.referenceFlagstatOutput(),
+                TestInputs.sageSomaticOutput(),
+                TestInputs.sageGermlineOutput(),
+                TestInputs.purpleOutput(),
+                TestInputs.chordOutput(),
+                TestInputs.lilacOutput(),
+                TestInputs.linxGermlineOutput(),
+                TestInputs.linxSomaticOutput(),
+                TestInputs.cuppaOutput(),
+                TestInputs.virusInterpreterOutput(),
+                TestInputs.peachOutput(),
+                TestInputs.sigsOutput(),
+                TestInputs.REF_GENOME_37_RESOURCE_FILES,
+                Pipeline.Context.RESEARCH);
+        List<BashCommand> commands = victim.tumorReferenceCommands(TestInputs.defaultSomaticRunMetadata());
+        assertThat(commands.get(2).asBash()).contains("-add_disclaimer");
```

Comment:
I don't love the use of the magic number, what about making it clear it's the last element using `commands.get(commands.size() - 1)`?

Reply:
One way or another it's a magic number. I am actually looking for the third and not the second to last. I could also pluck out the orange command by filtering on the orange jar or something but then am i really adding value? As a compromise I'll assign to a descriptive variable such that its clear for others what we're doing here